### ISSUE TITLE
RFI: HTTP Status Code and Error handling

### ISSUE THREAD

**Issue by ayoshi** [redodo/tortilla#7]
Currently, there is no convenient way to handle HTTP return codes and errors, this information is lost:

```
#
#... Skip...
 r = requests.request(method, url, params=params,
                             headers=request_headers, data=data, **kwargs)
#... Skip ...
        if has_body:
            return bunch.bunchify(json_response)
#... Skip ...
```
There has to be a way to handle errors.  There are several possible ways out of it, that I could think of:
1) Create a ```Response``` object to inherit from ```Bunch``` , with a _ResponseMixin_ that will define status_code method ( or property ) .
This has a disadvantage if you want to return user-defined objects with dump/load callable.

2) Put a property , like ```_status_code_```  on an object (of any type) before returning in. This has almost no disadvantages, besides some obscure case of return type already having ```_status_code_``` defined, in which case you can raise an exception

3) Have a method on an endpoint which will return last requests ```endpoint.last_status_code``` ( have to consider multiple instances with various async libs, when requests on the same endpoint are executed in parallel - but I didn't really think hard about this yet. )

Options (2), and (3) are probably better.

**Comment by hbredin** [comment]
What is the current status on this issue? 
I am willing to help if needed...

**Comment by redodo** [comment]
This is included in v0.4.0.
